spring-integration-redis/src/test/java/org/springframework/integration/redis/inbound/ReactiveRedisStreamMessageProducerTests.java

```diff
@@ -0,0 +1,166 @@
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.integration.redis.inbound;
+
+import static org.assertj.core.api.Assertions.assertThat;
+
+import org.junit.Before;
+import org.junit.Test;
+import org.junit.jupiter.api.AfterEach;
+import org.junit.runner.RunWith;
+
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.context.annotation.Bean;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.data.redis.connection.ReactiveRedisConnectionFactory;
+import org.springframework.data.redis.connection.stream.StreamInfo;
+import org.springframework.data.redis.core.ReactiveRedisTemplate;
+import org.springframework.data.redis.serializer.RedisSerializationContext;
+import org.springframework.integration.channel.FluxMessageChannel;
+import org.springframework.integration.config.EnableIntegration;
+import org.springframework.integration.dsl.IntegrationFlow;
+import org.springframework.integration.dsl.IntegrationFlows;
+import org.springframework.integration.dsl.MessageChannels;
+import org.springframework.integration.endpoint.MessageProducerSupport;
+import org.springframework.integration.handler.ReactiveMessageHandlerAdapter;
+import org.springframework.integration.redis.outbound.ReactiveRedisStreamMessageHandler;
+import org.springframework.integration.redis.rules.RedisAvailable;
+import org.springframework.integration.redis.rules.RedisAvailableRule;
+import org.springframework.integration.redis.rules.RedisAvailableTests;
+import org.springframework.integration.redis.util.Address;
+import org.springframework.integration.redis.util.Person;
+import org.springframework.messaging.support.GenericMessage;
+import org.springframework.test.annotation.DirtiesContext;
+import org.springframework.test.context.junit4.SpringRunner;
+
+import reactor.core.publisher.Flux;
+import reactor.test.StepVerifier;
+
+/**
+ * @author Attoumane Ahamadi
+ *
+ * @since 5.4
+ */
+@DirtiesContext
+@RunWith(SpringRunner.class)
+public class ReactiveRedisStreamMessageProducerTests extends RedisAvailableTests {
+
+	private static final String STREAM_KEY = "myStream";
+
+	private static final String CONSUMER = "consumer";
+
+	@Autowired
+	FluxMessageChannel fluxMessageChannel;
+
+	@Autowired
+	ReactiveRedisStreamMessageProducer redisStreamMessageProducer;
+
+	@Autowired
+	ReactiveRedisTemplate<String, ?> template;
+
+	@Autowired
+	ReactiveMessageHandlerAdapter messageHandler;
+
+	@Before
+	public void delKey() {
+		this.template.delete(STREAM_KEY).subscribe();
+	}
+
+	@AfterEach
```

Comment:
This is JUnit 5 feature. We can't use it with our Redis test suite yet because we stack there with JUnit 4 rules.
I mean something might be wrong with your tests since you combine JUnit 4 and 5.